charts/kong/templates/hpa.yaml

```diff
@@ -1,5 +1,9 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: {{ .Capabilities.APIVersions.Has "autoscaling/v2beta2" | ternary "autoscaling/v2beta2" "autoscaling/v1" }}
+{{- if .Capabilities.APIVersions.Has "autoscaling/v2" }}
+apiVersion: autoscaling/v2
+{{- else }}
+apiVersion: {{ (.Capabilities.APIVersions.Has "autoscaling/v2beta2" | ternary "autoscaling/v2beta2" "autoscaling/v1") }}
+{{- end}}
```

Comment:
Do you mind making a template out of this?

Reply:
Did you mean to make a template to decide the API version, and then call that template here?

Reply:
Yes

Reply:
Good idea, will do.

Reply:
done.